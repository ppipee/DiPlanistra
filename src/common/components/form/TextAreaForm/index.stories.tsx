import React from 'react'

import { Meta, Story } from '@storybook/react/types-6-0'

import TextAreaForm from '.'

export default {
	title: 'components/form/TextAreaForm',
	component: TextAreaForm,
} as Meta

const Template: Story = args => <TextAreaForm {...args} />

export const Example = Template.bind({})

Example.args = {
	variant: 'default',
	disabled: false,
}
