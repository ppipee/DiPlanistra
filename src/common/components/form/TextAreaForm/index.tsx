import React from 'react'

import TextAreaField from 'components/field/TextAreaField'
import { TextAreaFieldProps } from 'components/field/TextAreaField/types'
import FormTemplate from 'components/form/FormTemplate'
import { FormTemplateProps } from 'components/form/FormTemplate/types'

export type TextAreaFormProps = TextAreaFieldProps & FormTemplateProps

const TextAreaForm = ({ errMsg, label, ...props }: TextAreaFormProps) => (
	<FormTemplate variant={props.variant} errMsg={errMsg} label={label}>
		<TextAreaField as="textarea" {...props} />
	</FormTemplate>
)

export default TextAreaForm
