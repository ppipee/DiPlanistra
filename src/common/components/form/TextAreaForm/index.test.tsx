import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import TextAreaForm from '.'

describe('TextAreaForm', () => {
	it('should render TextAreaForm correctly', () => {
		const { container } = render(<TextAreaForm />)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render TextAreaForm with disabled state', () => {
		const { container } = render(<TextAreaForm disabled />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
