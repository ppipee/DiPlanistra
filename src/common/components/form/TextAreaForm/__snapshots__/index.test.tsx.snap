// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaForm should render TextAreaForm correctly 1`] = `
.c1 {
  width: 100%;
  border-radius: 4px;
  padding: 8px 12px;
  background-color: #FFFFFF;
  color: #263147;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  border: 1px solid #D3D6DA;
}

.c1::-webkit-input-placeholder {
  color: #7D8391;
}

.c1::-moz-placeholder {
  color: #7D8391;
}

.c1:-ms-input-placeholder {
  color: #7D8391;
}

.c1::placeholder {
  color: #7D8391;
}

.c1:focus {
  border: 1px solid #263147;
  box-shadow: 0 0 0 2px #D3D6DA;
}

.c1:disabled {
  color: #D3D6DA;
  background-color: #F7F7F8;
  border-color: #E9EAEC;
}

.c2 {
  height: 120px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<label
  class="c0"
>
  
  <textarea
    class="c1 c2"
  />
</label>
`;

exports[`TextAreaForm should render TextAreaForm with disabled state 1`] = `
.c1 {
  width: 100%;
  border-radius: 4px;
  padding: 8px 12px;
  background-color: #FFFFFF;
  color: #263147;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
  border: 1px solid #D3D6DA;
}

.c1::-webkit-input-placeholder {
  color: #7D8391;
}

.c1::-moz-placeholder {
  color: #7D8391;
}

.c1:-ms-input-placeholder {
  color: #7D8391;
}

.c1::placeholder {
  color: #7D8391;
}

.c1:focus {
  border: 1px solid #263147;
  box-shadow: 0 0 0 2px #D3D6DA;
}

.c1:disabled {
  color: #D3D6DA;
  background-color: #F7F7F8;
  border-color: #E9EAEC;
}

.c2 {
  height: 120px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<label
  class="c0"
>
  
  <textarea
    class="c1 c2"
    disabled=""
  />
</label>
`;
